src/Microsoft.AspNetCore.Mvc.Core/Internal/DefaultErrorDescriptorFactory.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Mvc.Abstractions;
+
+namespace Microsoft.AspNetCore.Mvc.ErrorDescription
+{
+    public class DefaultErrorDescriptorFactory : IErrorDescriptionFactory
```

Comment:
We have two levels of extensibility here: both 'factory' and within the factory, some 'providers'. Could you clarify why the two-tier system is necessary? I'm missing why it's not enough to have a single collection of providers.

Reply:
We have the pattern elsewhere in Mvc. Generally, you have a default provider and the user may append additional providers to run after to customize behavior. Doing so you let's you start from a default configured instance. The factory is meant to be a driver for the providers and we don't expect users to replace it.

Reply:
The factory is what our code calls. The providers are the extensibility